packages/client.api/src/object/ObjectMetadata.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+type ObjectTypeVisibility = "NORMAL" | "PROMINENT" | "HIDDEN";
+
+type BlueprintIcon = {
```

Comment:
If there is another type of icon like `StandardWindowsIcon` how will you know its any different from `BlueprintIcon`?

Reply:
Icon from gateway is just a union consisting of only the Blueprint type [shown here.](https://github.com/palantir/osdk-ts/blob/5a7443582c7e72acd0c389f840eba5813d1fe681/packages/internal.foundry.core/src/_components.ts#L680C1-L685C57) I was thinking we should keep parity with this for an easy update if it changes

Reply:
Agree but without the `type` field, if the `StandardWindowsIcon` interface also has the name and color, you won't know which to do